CenterForOpenScience/osf.io
# [IN-331][OSF] Contributor withdrawal emails

## Purpose

This PR enables automatic sending of emails to contributors upon preprint withdrawal.

## Changes

Implement the `notify_withdraw` hook of `ReviewsMachine`.

## QA Notes

Test to see whether email language is correct upon preprint withdrawal.
Two cases:
* When a contributor submits a withdrawal request and a moderator approves such request
  - An email should be sent to all contributors, including the request submitter. However the language is different for the request submitter.
* When a moderator withdraws a preprint
  - An email should be sent to all contributors. The language should say that a moderator performs such withdrawal.

For details on email language for different cases, see https://docs.google.com/document/d/19HwdNGkQ7jIrvEhZhLwQOpgvTXIJP22rjJp9EJfPGfU/edit

## Documentation



## Side Effects



## Ticket

https://openscience.atlassian.net/browse/IN-331
